### PR TITLE
Fix races on Node.Up field by encapsulation

### PR DESCRIPTION
**Pull request by frncmx**
The Node.Up field was accessed concurrently without "proper" locking.
There was a lock on Network and that was used sometimes to access
the  field. Other times the locking was missed and we had
a data race.

### CODE REVIEW

**Review by gluk256 — APPROVED** [review]
excellent job!

File: p2p/simulations/network.go [hunk 1]
```diff
@@ -595,8 +585,21 @@ type Node struct {
 	// Config if the config used to created the node
 	Config *adapters.NodeConfig `json:"config"`
 
-	// Up tracks whether or not the node is running
-	Up bool `json:"up"`
+	// up tracks whether or not the node is running
+	up   bool         `json:"up"`
```
Comment (janos) [inline p2p/simulations/network.go]: Unexported filed cannot be json-encoded, and encoding is handled in UnmarshalJSON. Json tags can be removed from this struct.

**Review by janos — APPROVED** [review]
LGTM. 👍

File: p2p/simulations/network.go (hunk 1 above)
Comment (frncmx) [reply p2p/simulations/network.go]: :+1: fixed

**Comment by frncmx** [comment]
opened upstream: https://github.com/ethereum/go-ethereum/pull/18976

**Pull request closed without merging**